drivers/vulkan/rendering_device_vulkan.cpp

```diff
@@ -8933,11 +8968,8 @@ void RenderingDeviceVulkan::initialize(VulkanContext *p_context, bool p_local_de
 	context = p_context;
 	device = p_context->get_device();
 	if (p_local_device) {
-		frame_count = 1;
```

Comment:
The implications of removing this aren't totally clear to me. The logic with p_local_device is slightly different as it isn't connected to a window in that case. I'll have to think about this more.

Reply:
Hi!

There are two independent things going on:

1. What frame_count was supposed to do. I've been meaning to ask you that. (and it sounds like this should be user-provided rather than assuming a value).
      - Right now `FRAME_LAG` does the job of the max frames to be ahead. It is hardcoded but I will be modifying this to be user-tweakable
2. `frame_count` was completely broken as I described in #80550 .

Reply:
The issue I am referring to comes from the fact that the local device doesn't have a swapchain and it never calls swap_buffers. It is essentially one frame, and it has to call ``submit()`` then ``sync()`` in order to submit more operations. In those cases, frame_count should always be 1 as there will literally never be more than one "frame".

Obtaining a global frame count from the context fundamentally breaks thing for local devices. 

I'm not opposed to grabbing the frame count from context, but if we do that, then we need more checks if we are using a local device to ensure we aren't referencing resources that don't exist